controllers/provider-aws/pkg/controlplane/actuator.go

```diff
@@ -0,0 +1,363 @@
+	if err := ccmChart.Delete(ctx, a.client, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete CCM objects for controlplane '%s'", objectName(cp))
+	}
+
+	// Delete config objects
+	a.logger.Info("Deleting configuration objects", "controlplane", objectName(cp))
+	if err := configChart.Delete(ctx, a.client, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete configuration objects for controlplane '%s'", objectName(cp))
+	}
+
+	// Delete secrets
+	a.logger.Info("Deleting secrets", "controlplane", objectName(cp))
+	if err := controlPlaneSecrets.Delete(a.clientset, cp.Namespace); err != nil {
+		return errors.Wrapf(err, "could not delete secrets for controlplane '%s'", objectName(cp))
+	}
+
+	return nil
+}
+
+// computeChecksums computes and returns all needed checksums. This includes the checksums for the given deployed secrets,
+// as well as the cloud provider secret and configmap that are fetched from the cluster.
+func (a *actuator) computeChecksums(
+	ctx context.Context,
+	deployedSecrets map[string]*corev1.Secret,
+	namespace string,
+) (map[string]string, error) {
+	// Get cloud provider secret and configmap from cluster
+	cpSecret := &corev1.Secret{}
+	err := a.client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: common.CloudProviderSecretName}, cpSecret)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get secret '%s'", objectName(cpSecret))
+	}
+	cpConfigMap := &corev1.ConfigMap{}
+	err = a.client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: common.CloudProviderConfigName}, cpConfigMap)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get configmap '%s'", objectName(cpConfigMap))
+	}
+
+	// Compute checksums
+	csSecrets := controlplane.MergeSecretMaps(deployedSecrets, map[string]*corev1.Secret{
+		common.CloudProviderSecretName: cpSecret,
+	})
+	csConfigMaps := map[string]*corev1.ConfigMap{
+		common.CloudProviderConfigName: cpConfigMap,
+	}
+	return controlplane.ComputeChecksums(csSecrets, csConfigMaps), nil
+}
+
+// getSubnetIDAndZone determines the subnet ID and zone from the given infrastructure status by looking for the first
+// subnet with purpose "public".
+func getSubnetIDAndZone(infraStatus *apisaws.InfrastructureStatus) (string, string, error) {
+	for _, subnet := range infraStatus.VPC.Subnets {
+		if subnet.Purpose == apisaws.PurposePublic {
+			return subnet.ID, subnet.Zone, nil
+		}
+	}
+	return "", "", errors.Errorf("subnet with purpose 'public' not found")
+}
+
+// getCloudProviderConfig builds and returns a AWS config from the given parameters.
+func getCloudProviderConfig(vpcID, subnetID, zone, clusterID string) string {
+	return fmt.Sprintf(
+		`[Global]
+VPC=%q
+SubnetID=%q
+DisableSecurityGroupIngress=true
+KubernetesClusterTag=%q
+KubernetesClusterID=%q
+Zone=%q
+`,
+		vpcID, subnetID, clusterID, clusterID, zone)
+}
+
+func objectName(obj runtime.Object) string {
+	k, _ := client.ObjectKeyFromObject(obj)
```

Comment:
Why are you ignoring the error here?

Reply:
Because it's a private method and I am quite sure an error can never occur. Also, in case it does occur, k would be an empty key that would be returned as "/". Since this method is only used in log entries, that would be not ideal, but fine. Returning the error from this method would complicate the client code too much with no benefit whatsoever.

Reply:
there are benefits, you don't risk unexpected behavior that are made harder to debug due to lack of err. Also as described in the documentation, the error is returned to handle this case:  

> is returned when an object implements the List style interfaces but not the Object style interfaces.

So it does not cause alot of trouble IMHO to not ignore the error.

Reply:
OK, so I added 3 more lines of code and the behaviour is exactly the same result as before. Hopefully the benefits are worth it.